### DISCUSSION TITLE
Apple TV / tvOS 17.0 support

### DISCUSSION THREAD

**Discussion by M4T1A5** [mullvad/mullvadvpn-app#8972] [Feature requests & Ideas]
## Issue report

Operating system: tvOS 17.0+

### Issue description

Apparently the recently released tvOS 17 for Apple TV boxes added support for third-party VPN apps (see for example [latest additions to developer SDK](https://developer.apple.com/whats-new/) or [Network Extensions documentation](https://developer.apple.com/documentation/networkextension)), so it would be nice if Mullvad could eventually add support for it.

I don't personally own any macOS devices so unfortunately I can't work on adding such support, but as the iOS app already exists, it seems like this would "only" need the UI part for tvOS.

**Comment by pinkisemils** [comment]
We have estimated the amount of work it'd take, and last time we checked, it was a bit more than just the UI part. We are of course optimistic about this and _eventually_ we'd like to release an app for tvOS, but it's not a near-term goal for us.
Unfortunately, none of the UI controls would work, and there were some hoops to get golang and Rust to work with tvOS when we last toyed with this idea.

**Comment by zeman** [comment]
Looks like a lot of VPN apps are starting to add tvOS support. Would be great to see Mullvad on this list... https://9to5mac.com/2023/11/25/best-vpn-apps-apple-tv/

**Comment by CaioBianchi** [comment]
I should've signed up via Tailscale. Tailscale has a functioning tvOS app, and they support Mullvad, but only for accounts created through their integration.

**Comment by thekeymaker33** [comment]
Many providers have added the support for tvOS app.

come on Mullvad.

**Comment by ehogeweg** [comment]
> Many providers have added the support for tvOS app.
> 
> come on Mullvad.

+1

**Comment by Clickbaitcake** [comment]
> > Many providers have added the support for tvOS app.
> > come on Mullvad.
> 
> +1

+1

**Comment by AceOf5pades** [comment]
> > > Many providers have added the support for tvOS app.
> > > come on Mullvad.
> > 
> > 
> > +1
> 
> +1

+1

**Comment by clouddistortion** [comment]
I am in if that app is available for AppleTV

**Comment by kingkong404** [comment]
Any update here?

**Comment by thekeymaker33** [comment]
how is it going?

**Comment by Spl0itable** [comment]
Where are we at with this? Nord, Express, and Proton now all have tvOS apps.

**Comment by tobsn** [comment]
*cough

**Comment by voidedluvsu** [comment]
*sneeze

**Comment by metametapod** [comment]
FWIW, it’s possible to get this working via downloaded Mullvad WireGuard or OpenVPN profiles and a generic VPN client like [Passepartout](https://passepartoutvpn.app/).

**Comment by mstaack** [comment]
or you can use tailscale app on tvos with mullvad exit node https://tailscale.com/kb/1258/mullvad-exit-nodes

**Comment by thekeymaker33** [comment]
Or Mullvad could create an app for Apple TV like all other vpn providers that have done it.It shouldn’t be that hard to make a universal app that works on tvOS, iOS, iPadOS.Sent from my iPhoneOn 2 Oct 2025, at 16:45, Max ***@***.***> wrote:﻿mstaack left a comment (mullvad/mullvadvpn-app#5279)
or you can use tailscale and and mullvad exit node https://tailscale.com/kb/1258/mullvad-exit-nodes

**Comment by tobsn** [comment]
> or you can use tailscale app on tvos with mullvad exit node https://tailscale.com/kb/1258/mullvad-exit-nodes

> FWIW, it’s possible to get this working via downloaded Mullvad WireGuard or OpenVPN profiles and a generic VPN client like [Passepartout](https://passepartoutvpn.app/).


everyone knows about passepartout and tailscale. this specific issue is about a tvOS app, not workarounds.

**Comment by thekeymaker33** [comment]
No thanks!

![Image](https://github.com/user-attachments/assets/122c4188-6321-4942-9f04-8f4422125faf)

**Comment by git849376** [comment]
Any update here?

**Comment by antonmartinsson** [comment]
Still no updates? Would love to switch to Mullvad but no tvOS app is a dealbreaker for me.

**Reply by thekeymaker33** [reply]
I got tired of waiting and bought a GL.iNet router.

but I don’t like the fact that nobody from mullvad write any comments here.

**Reply by antonmartinsson** [reply]
I'm sadly not techy enough to bother doing anything other than just installing an app. 

Looks like I might have to go with Nord for another year. 😓

**Reply by Spl0itable** [reply]
I'd highly recommend Proton over Nord lol

**Reply by pinkisemils** [reply]
We have plans to attempt to deliver a minimal app towards the end of the year. Wouldn't want to promise that we will definitely have an _app store_ ready app by the end of the year, but we will definitely work on one this year.

**Reply by antonmartinsson** [reply]
Thanks for the update @pinkisemils! 🙌
